### DISCUSSION TITLE
Add player & onlinePlayer check to specialized command check:[value1]==[value2]

### DISCUSSION THREAD

**Discussion by DrakePork** [Zrips/CMI#4734] [Feature Requests]
**Description of feature/improvement request:**
Basically add two new possible checks for the check: in specialized commands, which is to check for if $1 is a valid player, and o check if its a valid online player.

---
### Detailed explanation of what it should do, when and in what situation: ###

If I have
- check:$1==onlinePlayer! 

then if $1 is a valid player and the player is also online, then it should proceed, and if I have
- check: $1==player! 

then it should proceed if its a valid player whether or not the player is online or offline.

### Basic explanation as to why this feature is needed and/or useful to have: ###
Would be very useful for if you've got a custom alias that has a command that requires a valid player as input.

**Comment by CloudeLecaw** [comment]
You have

> ifonline [playerName] (command)
> Perform command only when player is online

And you can design a check like `ifonline:[playerName]!`

> If command starts with ifonline:[playerName]! then command will get performed if player is online. Static name or [playerName] can be depending what result you want to get.

check:$1==onlinePlayer! <- does this makes sense?
you should write your examle where u tried this check.
.
.
about check: $1==player!

i dont think this is a real usefull check.. i see no reason where the command should check if the player exist or not. This could result in a few 10000 checks on bigger servers each time...
